src/Uno.UI.Composition/Composition/CompositionSurfaceBrush.skia.cs

```diff
@@ -87,6 +88,21 @@ internal override void UpdatePaint(SKPaint fillPaint, SKRect bounds)
 				}
 				else
 				{
+					var matrix = Matrix3x2.CreateTranslation((float)backgroundArea.Left, (float)backgroundArea.Top);
+					matrix *= TransformMatrix;
+
+					// The brush is not tied to a specific window, so we can't get the scaling of a specific XamlRoot and
+					// instead we settle for the main window's scaling which should hopefully be the same.
+					var scale = DisplayInformation.GetForCurrentViewSafe().LogicalDpi / DisplayInformation.BaseDpi;
+					// We scale the backgroundArea with the dpi scaling and then scale with 1 / dpi scaling in the shader,
+					// This is conceptually a noop, but it actually affects quality. For example, if you have a hi-res image
+					// that should be drawn in a 500x500 rectangle. If the scaling is 2x, it will actually be drawn in a
+					// 1000x1000 rectangle. We don't want to resize the image to 500x500 and then stretch it to 1000x1000.
+					// Instead, we resize it to 1000x1000 directly and counteract the global 2x scaling done by the
+					// renderer by telling the shader to scale by 1/2.
+					backgroundArea = backgroundArea with { Width = backgroundArea.Width * scale, Height = backgroundArea.Height * scale };
+					matrix *= Matrix3x2.CreateScale(1 / scale);
```

Comment:
Can you add a manual test for this, asking that the app needs to be run in 2x to validate the behavior?